dev-daramir/ dashboard: tBTC falling auction participation page

daramir opened the pull request:
# Description
- LiquidationsPage (the page with events that could lead to liquidation) now captures the GotRedemptionSignature and CourtesyCalled. This means it now captures all event sources, finds the related deposit and shows it's status.
- This PR also contains an inclusion of yet another page in the Token Dashboard. Which would be the continuation of the liquidation flow, with the possibility to participate in falling price auction and purchase ETH from tBTC.

In my view, this work would greatly improve in the short future with:
- (In LiquidationsPage), filtering features, such as, actionable events only, special type of event, dates, operators (only mine etc),
- better styling
- possibly web alerts for new events in LiquidationsPage (and LiquidateDepositPage)
#### Credits
For this I heavily relied Rafal C's dashboard work, elishadrion's liquidator repo and https://github.com/keep-network/tbtc.js for `purchaseSignerBondsAtAuction` functionality:  - thank you

Please leave your feedback!


## Checklist
- [X] Fork keep-network/keep-core
- [X] Clone your fork
- [X] Describe the change you are intending to undertake in the PR description.

Before marking the PR as ready for review, make sure:
- [ ] It passes the linter checks (npm run lint) (see Pre-commit to make this automatic).
- [ ] It passes the continuous integration tests on CircleCI.
I don't have access to this CircleCI Org
- [ ] Your changes have sufficient test coverage (e.g regression tests have been added for bug fixes, unit tests for new features)
- [X] Commits must be signed.

daramir committed:
Merge pull request #1 from keep-network/master

upstream sync

daramir committed:
Merge pull request #2 from keep-network/master

upstream sync

daramir committed:
Merge pull request #3 from keep-network/master

upstream sync

daramir committed:
Merge pull request #4 from keep-network/master

upstream sync

daramir committed:
Update @keep-network/* dashboard dependencies
match latest contracts on ropsten

daramir committed:
ran `npm audit fix`
fixed 6388 of 6477 vulnerabilities in 3522 scanned packages
  89 vulnerabilities required manual review and could not be updated

daramir committed:
Ropsten Readme Instructions
Connecting to Ropsten from localhost instructions in Readme

Signed-off-by: Danilo Ramirez <daramir9999@gmail.com>

daramir committed:
Add Deposit status constants and tBtcViaWeitoshi helper

daramir committed:
npm js:lint:fix constants

daramir committed:
useSubscribeToContractEvent error handling

daramir committed:
Inclusion of tbtc-liquidation.service.js

daramir committed:
Missed adding contractName to dependency array
(gave warning)

daramir committed:
Fixed warnings in tbtc-liquidation.service

daramir committed:
All about Liquidations functionality

daramir committed:
Merge pull request #5 from keep-network/master

upstream sync

daramir committed:
Merge branch 'master' into dev-daramir/dashboard-liquidations

daramir committed:
Add support for CourtesyCalled and GotRedemptionSignature events

daramir committed:
Add functionality for consumption of TBTCConstants

daramir committed:
Allow cellStyle in DataTable Component

daramir commented:
Sample screenshots of how this looks like

On liquidations page....
![LiquidationsPage](https://user-images.githubusercontent.com/4935829/91792300-8cb26480-ec58-11ea-830d-4d0bf9f16362.png)

After clicking "Liquidate" and "View Liquidation"
![After clicking view liquidation](https://user-images.githubusercontent.com/4935829/91792332-a5227f00-ec58-11ea-9ee4-d0827c69aa35.png)

After a few seconds (it has periodic fetch) or page refresh
![image](https://user-images.githubusercontent.com/4935829/91791968-cc2c8100-ec57-11ea-8c5c-aa624e65ee84.png)

If URL is visited for a deposit not in liquidation
![image](https://user-images.githubusercontent.com/4935829/91792021-e5cdc880-ec57-11ea-8aea-1b06055b63b1.png)

daramir committed:
Support routing for new hub page

daramir committed:
Add hub page

daramir committed:
Add style file for new page
./solidity/dashboard/src/css/liquidate-deposit-page.less - Will need improvement

daramir committed:
Add Link from TimelineElement to Auction page
Will need logic improvement to not always show

daramir committed:
Adding new page (LiquidateDepositPage.jsx) and dependencies
They all need cleanup

daramir committed:
Tiny fixes for DepositLiquidationOverview
Show nicer modal
Purchase from auctoin logic and aution page logic

daramir committed:
Merge pull request #6 from keep-network/master

upstream sync

daramir committed:
Merge pull request #7 from keep-network/master

upstream sync

daramir committed:
Merge branch 'master' into dev-daramir/dashboard-liquidations

daramir committed:
Merge branch 'dev-daramir/dashboard-liquidations' into dev-daramir/dashboard-liquidations-update-1

# Conflicts:
#	solidity/dashboard/src/constants/constants.js
#	solidity/dashboard/src/contracts.js

daramir committed:
Fix variable name

daramir committed:
Get TBTCTokenContract the new way
Still need to adapt rest of the file to new pattern

mhluongo commented:
@daramir this is awesome- @liz-shinn @r-czajkowski thoughts on this? Mostly interested in how best to integrate it in the dashboard plan

daramir committed:
Merge pull request #8 from keep-network/master

upstream sync

daramir committed:
Merge branch 'master' into dev-daramir/dashboard-liquidations

daramir committed:
Fix arbitrageur tBTC balance not showing fully
display function was rounding down to 0 decimals

daramir committed:
purchaseDepositAtAuction improvements
Bring back gasEstimate
Fix confirmation message (had static values *smh*)
Swap auctionValue for `lotSizeTbtc` call

daramir committed:
Merge pull request #9 from keep-network/master

upstream sync

daramir committed:
Merge branch 'master' into dev-daramir/dashboard-liquidations

daramir committed:
Merge branch 'dev-daramir/dashboard-liquidations' into dev-daramir/dashboard-liquidations-update-1

daramir committed:
Merge pull request #10 from keep-network/master

upstream sync

daramir committed:
Merge branch 'master' into dev-daramir/dashboard-liquidations

daramir committed:
Merge branch 'dev-daramir/dashboard-liquidations' into dev-daramir/dashboard-liquidations-update-1

daramir committed:
Merge pull request #11 from keep-network/master

upstream sync

daramir committed:
Merge branch 'master' into dev-daramir/dashboard-liquidations

daramir committed:
Merge branch 'dev-daramir/dashboard-liquidations' into dev-daramir/dashboard-liquidations-update-1

benlongstaff commented:
this looks awesome

daramir commented:
Hi @benlongstaff,

I have a standalone version coming up while the PR goes through (I'm still
actively updating it anyway)

Will let you know when the standalone is publicly available. Just want to
get a thumbs up from the team on a few details.

Cheers

On Thu, Oct 22, 2020, 1:58 AM benlongstaff <notifications@github.com> wrote:

> this looks awesome
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keep-network/keep-core/pull/2027#issuecomment-713639130>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABFVBFJHAGAYQCFVNA4D7ZTSL3ZKPANCNFSM4QRGWB5Q>
> .
>
